src/nvim/api/buffer.c

```diff
@@ -179,6 +179,11 @@ void buffer_set_line_slice(Buffer buffer,
     return;
   }
 
+  if (!inbounds(buf, start)) {
```

Comment:
I'm not sure. While it would certainly be convenient, it's not consistent with vim commands like `setline` (if we're trying to emulate those) and it might be confusing for the behavior of `set_line_slice` to be entirely different depending on which end of the buffer has been overrun.

At least from my perspective, it would be much more common for an out-of-bounds set to result from plugin/user race conditions than from trying to set lines at a specific position in a file. I'd be interested in hearing of others' experiences, though.

It might just be easier and more explicit to just send over a chunk of vimscript/lua that performs the appropriate operations.